src/nica/nc-string.c

```diff
@@ -109,6 +109,27 @@ bool nc_string_cat(nc_string *s, const char *append)
         return true;
 }
 
+bool nc_string_prepend(nc_string *s, const char *prepend)
+{
+        char *p = NULL;
+        int len = 0;
+
+        if (!s || !prepend) {
+                return false;
+        }
+        if (!s->str) {
+                return false;
+        }
+        len = asprintf(&p, "%s%s", prepend, s->str);
+        if (!p || len < s->len) {
+                return false;
+        }
+        free(s->str);
+        s->str = p;
+        s->len = len;
+        return true;
+}
```

Comment:
Can you submit a PR to upstream libnica to add this function? I would rather not use a forked version of the library.